### PR TITLE
feat(sbb-checkbox, sbb-radio-button): split into regular and panel variants

### LINKED ISSUE #2395: Separate checkbox and radio buttons into base components and panel components
sbb-checkbox should be separated into sbb-checkbox and sbb-checkbox-panel (which inherits from sbb-checkbox)
sbb-radio-button should be separated into sbb-radio-button and sbb-radio-button-panel (which inherits from sbb-radio-button)

For both this means moving/extracting the panel specific slots into the panel variants.
Additionally the sbb-selection-panel integration should be adapted for the panel variants and removed for the base variants.
The panel components should work well with the sbb-checkbox-group and sbb-radio-button-group respectively.

Pay special attention to styling for border with standalone panels and within selection panel.

```html
<sbb-checkbox>My checkbox</sbb-checkbox>

<sbb-checkbox-panel>
  My checkbox panel
  <span slot="subtext">My subtext</span>
</sbb-checkbox-panel>

<sbb-selection-expansion-panel>
  <sbb-checkbox-panel>
    My checkbox panel
    <span slot="subtext">My subtext</span>
  </sbb-checkbox-panel>
  <div slot="content">
    My content
  </div>
</sbb-selection-expansion-panel>
```

### PR DESCRIPTION
**Pull request by MarioCastigliano**
Closes #2395 

BREAKING CHANGE: `sbb-selection-panel` has been renamed to `sbb-selection-expansion-panel`. The `sbb-checkbox` and `sbb-radio-button` components cannot be used anymore with `sbb-selection-expansion-panel` (does not apply for cases where they are slotted inside the `content` slot). As a replacement, we introduce the new components `sbb-checkbox-panel` and `sbb-radio-button-panel`, which could also be used standalone in cases where there is no content. `sbb-checkbox-group` and `sbb-radio-button-group` also support the panel variants.
How to migrate?
- Rename usages of `sbb-selection-panel` to `sbb-selection-expansion-panel`.
- Inside the `sbb-selection-expansion-panel`, replace `sbb-checkbox` with `sbb-checkbox-panel` and `sbb-radio-button` with `sbb-radio-button-panel` (does not apply for cases where they are slotted inside the `content` slot of the `sbb-selection-expansion-panel`)
- In cases where there was no content (slot), don't use `sbb-selection-panel`/`sbb-selection-expansion-panel` anymore, but directly use `sbb-checkbox-panel` or `sbb-radio-button-panel`.

### CODE REVIEW

File: src/components/checkbox/checkbox-panel/checkbox-panel.ts [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+  checkboxPanelStyle,
+  commonStyle,
+  type SbbCheckboxStateChange,
+} from '../common/index.js';
+
+import '../../screen-reader-only/index.js';
+import '../../visual-checkbox/index.js';
+
+/**
+ * Describe the purpose of the component with a single short sentence.
```
Comment (DavideMininni-Fincons) [inline src/components/checkbox/checkbox-panel/checkbox-panel.ts:26]: Need the real component's description

File: src/components/checkbox/common/checkbox-panel.scss [hunk 2]
```diff
@@ -0,0 +1,94 @@
+@use '../../core/styles' as sbb;
```
Comment (DavideMininni-Fincons) [inline src/components/checkbox/common/checkbox-panel.scss:1]: Is this used only in checkbox-panel-component? if yes, it can be moved there.

File: src/components/checkbox/common/checkbox.scss [hunk 3]
```diff
@@ -0,0 +1,75 @@
+@use '../../core/styles' as sbb;
```
Comment (DavideMininni-Fincons) [inline src/components/checkbox/common/checkbox.scss:1]: Is this used only in checkbox component? if yes, it can be moved there.

File: src/components/checkbox/checkbox-panel/checkbox-panel.ts [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+    this.toggleAttribute(
+      'data-is-selection-panel-input',
+      !!this._selectionPanelElement &&
+        !this.closest?.('sbb-selection-expansion-panel [slot="content"]'),
+    );
+
+    this._checkboxLoaded.emit();
+
+    // We need to call requestUpdate to update the reflected attributes
+    ['disabled', 'required', 'size'].forEach((p) => this.requestUpdate(p));
```
Comment (DavideMininni-Fincons) [inline src/components/checkbox/checkbox-panel/checkbox-panel.ts:101]: 1. Does the panel has a `size`?
2. this line is duplicated between here and the checkbox

File: src/components/checkbox/checkbox/checkbox.ts [hunk 5]
```diff
@@ -1,71 +1,32 @@
...
  * @event {CustomEvent<void>} didChange - Deprecated. used for React. Will probably be removed once React 19 is available.
  * @event {Event} change - Event fired on change.
  * @event {InputEvent} input - Event fired on input.
  */
 @customElement('sbb-checkbox')
-export class SbbCheckboxElement extends SbbUpdateSchedulerMixin(
-  SbbFormAssociatedCheckboxMixin(SbbIconNameMixin(SbbHydrationMixin(LitElement))),
-) {
-  public static override styles: CSSResultGroup = style;
+export class SbbCheckboxElement extends SbbCheckboxCommonElementMixin(LitElement) {
```
Comment (DavideMininni-Fincons) [inline src/components/checkbox/checkbox/checkbox.ts:23]: `events` are shared with the panel (which has 1 more); can it be moved in the common file?
same for the eventemitters (eg. _checkboxLoaded)

File: src/components/checkbox/checkbox-group/checkbox-group.stories.ts [hunk 6]
```diff
@@ -48,6 +49,38 @@ const checkboxes = (
...
+  disabledSingle: boolean,
+  iconName: string,
+  iconPlacement: string,
+  label: string,
+): TemplateResult => html`
+  <sbb-checkbox-panel
+    value="checkbox-1"
+    ?checked=${checked}
+    icon-name=${iconName || nothing}
+    icon-placement=${iconPlacement}
```
Comment (DavideMininni-Fincons) [inline src/components/checkbox/checkbox-group/checkbox-group.stories.ts:63]:
```suggestion
    icon-placement=${iconPlacement || nothing}
```

File: src/components/checkbox/checkbox-group/checkbox-group.stories.ts [hunk 7]
```diff
@@ -48,6 +49,38 @@ const checkboxes = (
...
+  >
+    ${label} 1
+  </sbb-checkbox-panel>
+  <sbb-checkbox-panel
+    value="checkbox-2"
+    ?disabled=${disabledSingle}
+    icon-name=${iconName || nothing}
+    icon-placement=${iconPlacement}
```
Comment (DavideMininni-Fincons) [inline src/components/checkbox/checkbox-group/checkbox-group.stories.ts:71]:
```suggestion
    icon-placement=${iconPlacement || nothing}
```

File: src/components/checkbox/checkbox-group/checkbox-group.stories.ts [hunk 8]
```diff
@@ -48,6 +49,38 @@ const checkboxes = (
...
+  >
+    ${label} 2
+  </sbb-checkbox-panel>
+  <sbb-checkbox-panel
+    value="checkbox-3"
+    icon-name=${iconName || nothing}
+    icon-placement=${iconPlacement}
```
Comment (DavideMininni-Fincons) [inline src/components/checkbox/checkbox-group/checkbox-group.stories.ts:78]:
```suggestion
    icon-placement=${iconPlacement || nothing}
```

File: src/components/checkbox/checkbox-group/checkbox-group.ts [hunk 9]
```diff
@@ -1,5 +1,5 @@
 import type { CSSResultGroup, PropertyValues, TemplateResult } from 'lit';
-import { html, LitElement } from 'lit';
+import { LitElement, html } from 'lit';
```
Comment (DavideMininni-Fincons) [inline src/components/checkbox/checkbox-group/checkbox-group.ts:2]: Why? :D

File: src/components/core/mixins/form-associated-checkbox-mixin.ts [hunk 10]
```diff
@@ -81,6 +81,7 @@ export const SbbFormAssociatedCheckboxMixin = <T extends Constructor<LitElement>
       }
       this._checked = Boolean(value);
       this.updateFormValue();
+      this.toggleAttribute('data-checked', this._checked);
```
Comment (DavideMininni-Fincons) [inline src/components/core/mixins/form-associated-checkbox-mixin.ts:84]: I would prefer:

```suggestion
      this._checked = Boolean(value);
      this.toggleAttribute('data-checked', this._checked);
      this.updateFormValue();
```

**Review by DavideMininni-Fincons — CHANGES_REQUESTED** [review]
Partial review (checkbox)

File: src/components/radio-button/radio-button/radio-button.ts [hunk 11]
```diff
@@ -1,110 +1,32 @@
-import type { CSSResultGroup, PropertyValues, TemplateResult } from 'lit';
-import { html, LitElement, nothing } from 'lit';
-import { customElement, property, state } from 'lit/decorators.js';
+import type { CSSResultGroup, TemplateResult } from 'lit';
+import { LitElement, html, nothing } from 'lit';
```
Comment (DavideMininni-Fincons) [inline src/components/radio-button/radio-button/radio-button.ts:2]: Same here - why inversion?

File: src/components/radio-button/radio-button-panel/radio-button-panel.ts [hunk 12]
```diff
@@ -0,0 +1,164 @@
...
+    this._isSelectionPanelInput =
+      !!this._selectionPanelElement &&
+      !this.closest('sbb-selection-expansion-panel [slot="content"]');
+
+    this.toggleAttribute('data-is-selection-panel-input', this._isSelectionPanelInput);
+
+    this._radioButtonLoaded.emit();
+
+    // We need to call requestUpdate to update the reflected attributes
+    ['disabled', 'required', 'size'].forEach((p) => this.requestUpdate(p));
```
Comment (DavideMininni-Fincons) [inline src/components/radio-button/radio-button-panel/radio-button-panel.ts:108]: Similar to checkbox - does it have a size prop?

File: src/components/radio-button/radio-button-panel/radio-button-panel.spec.ts [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+
+  beforeEach(async () => {
+    element = await fixture(
+      html`<sbb-radio-button-panel value="radio-value">
+        Label
+        <span slot="subtext">Subtext</span>
+        <span slot="suffix">Suffix</span>
+      </sbb-radio-button-panel>`,
+    );
+  });
```
Comment (DavideMininni-Fincons) [inline src/components/radio-button/radio-button-panel/radio-button-panel.spec.ts:40]: Do i miss something? what's the difference with previous case?

File: src/components/radio-button/radio-button/radio-button.ts [hunk 14]
```diff
@@ -1,110 +1,32 @@
...
 @customElement('sbb-radio-button')
-@hostAttributes({
-  role: 'radio',
-})
-export class SbbRadioButtonElement extends SbbUpdateSchedulerMixin(LitElement) {
-  public static override styles: CSSResultGroup = style;
+export class SbbRadioButtonElement extends SbbRadioButtonCommonElementMixin(LitElement) {
+  public static override styles: CSSResultGroup = [commonStyle, radioButtonStyle];
   public static readonly events = {
     stateChange: 'stateChange',
```
Comment (DavideMininni-Fincons) [inline src/components/radio-button/radio-button/radio-button.ts:26]: Similar to checkbox - is possibile to move events/eventemitters in common?

File: src/components/radio-button/radio-button-group/radio-button-group.ts [hunk 15]
```diff
@@ -1,5 +1,5 @@
 import type { CSSResultGroup, PropertyValues, TemplateResult } from 'lit';
-import { html, LitElement } from 'lit';
+import { LitElement, html } from 'lit';
```
Comment (DavideMininni-Fincons) [inline src/components/radio-button/radio-button-group/radio-button-group.ts:2]: We meet again..

**Review by DavideMininni-Fincons — CHANGES_REQUESTED** [review]
Partial review (radio-button)

File: src/components/checkbox/checkbox-group/checkbox-group.ts (hunk 9 above)
Comment (MarioCastigliano) [reply src/components/checkbox/checkbox-group/checkbox-group.ts:2]: VS Code import auto-format sometimes moves things around like this 😅

File: src/components/radio-button/common/radio-button-common.ts [hunk 16]
```diff
@@ -0,0 +1,157 @@
...
+  protected handleDisabledChange(currentValue: boolean, previousValue: boolean): void;
+}
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export const SbbRadioButtonCommonElementMixin = <T extends AbstractConstructor<LitElement>>(
+  superClass: T,
+): AbstractConstructor<SbbRadioButtonCommonElementMixinType> & T => {
+  @hostAttributes({
+    role: 'radio',
+  })
```
[the remark this answers is not in the document]
Comment (MarioCastigliano) [reply src/components/radio-button/common/radio-button-common.ts:39]: Apparently it is 😅

File: src/components/radio-button/radio-button-panel/radio-button-panel.spec.ts (hunk 13 above)
Comment (MarioCastigliano) [reply src/components/radio-button/radio-button-panel/radio-button-panel.spec.ts:40]: Messed up while copy-pasting, this should've been the checked case. Fixing it

File: src/components/checkbox/checkbox-panel/checkbox-panel.stories.ts [hunk 17]
```diff
@@ -0,0 +1,180 @@
...
+  indeterminate: false,
+  disabled: false,
+  label: 'Label',
+  value: 'Value',
+  name: 'name',
+  'aria-label': undefined,
+};
+
+const Template = ({ label, checked, ...args }: Args): TemplateResult =>
+  html` <sbb-checkbox-panel .checked=${checked} ?checked=${checked} ${sbbSpread(args)}>
```
Comment (DavideMininni-Fincons) [inline src/components/checkbox/checkbox-panel/checkbox-panel.stories.ts:78]: Why the duplicate checked prop (? and .)?

File: src/components/checkbox/checkbox-panel/readme.md [hunk 18]
```diff
@@ -0,0 +1,86 @@
...
+(e.g., a parent which is checked only if all the children are checked, otherwise is in indeterminate state).
+Clicking on a `sbb-checkbox-panel` in this state sets `checked` to `true` and `indeterminate` to false.
+
+```html
+<sbb-checkbox-panel value="indeterminate-checkbox-panel" indeterminate="true"
+  >Indeterminate state</sbb-checkbox-panel
+>
+```
+
+The component can be disbaled by using the `disabled` property.
```
Comment (DavideMininni-Fincons) [inline src/components/checkbox/checkbox-panel/readme.md:34]:
```suggestion
The component can be disabled by using the `disabled` property.
```

File: src/components/radio-button/common/radio-button-common.ts [hunk 19]
```diff
@@ -0,0 +1,157 @@
...
+    }
+
+    private _handleKeyDown(evt: KeyboardEvent): void {
+      if (evt.code === 'Space') {
+        this.select();
+      }
+    }
+
+    protected abstract handleCheckedChange(currentValue: boolean, previousValue: boolean): void;
+    protected abstract handleDisabledChange(currentValue: boolean, previousValue: boolean): void;
```
Comment (DavideMininni-Fincons) [inline src/components/radio-button/common/radio-button-common.ts:152]: For better visibility, I would like to have those lines right after the private props (_handlerRepository).

**Review by DavideMininni-Fincons — CHANGES_REQUESTED** [review]
Great work! some minor details to fix, then is approvable

File: src/components/checkbox/checkbox-panel/checkbox-panel.stories.ts (hunk 17 above)
Comment (MarioCastigliano) [reply src/components/checkbox/checkbox-panel/checkbox-panel.stories.ts:78]: I think it's needed to make the checkbox behave just like the native one, Jeri and Lukas mentioned the fact that they needed to jump through some hoops to achieve that

**Review by DavideMininni-Fincons — APPROVED** [review]

File: src/components/core/mixins/form-associated-checkbox-mixin.ts [hunk 20]
```diff
@@ -80,6 +80,7 @@ export const SbbFormAssociatedCheckboxMixin = <T extends Constructor<LitElement>
         this._attributeMutationBlocked = true;
       }
       this._checked = Boolean(value);
+      this.toggleAttribute('data-checked', this._checked);
```
Comment (jeripeierSBB) [inline src/components/core/mixins/form-associated-checkbox-mixin.ts:83]: Why is this needed? Can we avoid this?

File: src/components/radio-button/radio-button/radio-button.ts [hunk 21]
```diff
@@ -142,117 +50,32 @@ export class SbbRadioButtonElement extends SbbUpdateSchedulerMixin(LitElement) {
...
-      !!this._selectionPanelElement && !this.closest('sbb-selection-panel [slot="content"]');
-    this.toggleAttribute('data-is-selection-panel-input', this._isSelectionPanelInput);
-
-    const signal = this._abort.signal;
-    this.addEventListener('click', (e) => this._handleClick(e), { signal });
-    this.addEventListener('keydown', (e) => this._handleKeyDown(e), { signal });
-    this._handlerRepository.connect();
-    this._radioButtonLoaded.emit();
 
     // We need to call requestUpdate to update the reflected attributes
```
Comment (jeripeierSBB) [inline src/components/radio-button/radio-button/radio-button.ts:75]: Should probably land in the mixin?

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]
It's not a complete review, but checked most parts :)
Thanks for your work! I feel like the work not completely finished / checked, thanks for considering my comments and also general comments. I'm not sure wether all the logic is at the right place (e.g. in panel instead of selection panel or vice versa).

- focus outline broken: https://lyne-components-preview-pr2552.app.sbb.ch/?path=/story/components-sbb-radio-button-sbb-radio-button-group--horizontal
- broken example: https://lyne-components-preview-pr2552.app.sbb.ch/?path=/story/components-sbb-radio-button-sbb-radio-button-panel--default
- not sure whether these simple examples show the real-world scenario. so maybe enhance the examples with more content or ask @mcilurzo what he considers as a good example? ( https://lyne-components-preview-pr2552.app.sbb.ch/?path=/story/components-sbb-radio-button-sbb-radio-button-group--horizontal-panels)
- wrong focus outline: https://lyne-components-preview-pr2552.app.sbb.ch/?path=/story/components-sbb-selection-expansion-panel--with-checkbox
- not focusable: https://lyne-components-preview-pr2552.app.sbb.ch/?path=/story/components-sbb-selection-expansion-panel--with-radio-button
- animation missing for selection panel (height, border-width, and border-color)
- broken stories: https://lyne-components-preview-pr2552.app.sbb.ch/?path=/story/components-sbb-selection-expansion-panel--with-no-content,  https://lyne-components-preview-pr2552.app.sbb.ch/?path=/story/components-sbb-selection-expansion-panel--with-no-content-checked-disabled
- from a design perspective, nested panels are not allowed, please achieve like it looked before: https://lyne-components-preview-pr2552.app.sbb.ch/?path=/story/components-sbb-selection-expansion-panel--nested-radios

File: src/components/core/mixins/form-associated-checkbox-mixin.ts (hunk 20 above)
Comment (MarioCastigliano) [reply src/components/core/mixins/form-associated-checkbox-mixin.ts:83]: selection-panel used `data-checked` for styling, in this case we didn't have anything on the component itself (for standalone use) and since we adopted the logic from the native checkbox the `checked` attribute is not suitable for styling the component. I couldn't come up with anything else, but if you have suggestions let me know 😅

File: src/components/checkbox/checkbox/checkbox.ts [hunk 22]
```diff
@@ -76,46 +35,9 @@ export class SbbCheckboxElement extends SbbUpdateSchedulerMixin(
...
-  private _checkboxLoaded: EventEmitter<void> = new EventEmitter(
-    this,
-    SbbCheckboxElement.events.checkboxLoaded,
-    { bubbles: true },
-  );
+  /** The label position relative to the labelIcon. Defaults to end */
+  @property({ attribute: 'icon-placement', reflect: true })
+  public iconPlacement: SbbIconPlacement = 'end';
 
   public constructor() {
```
[the remark this answers is not in the document]
Comment (MarioCastigliano) [reply src/components/checkbox/checkbox/checkbox.ts:42]: Panel version doesn't support icon. Or at lease that was my understanding of it, am I missing something? 🤔

File: src/components/checkbox/checkbox/checkbox.ts [hunk 23]
```diff
@@ -124,77 +46,13 @@ export class SbbCheckboxElement extends SbbUpdateSchedulerMixin(
...
-    this.toggleAttribute(
-      'data-is-selection-panel-input',
-      !!this._selectionPanelElement && !this.closest?.('sbb-selection-panel [slot="content"]'),
-    );
 
-    this._checkboxLoaded.emit();
+    this.checkboxLoaded.emit();
 
     // We need to call requestUpdate to update the reflected attributes
     ['disabled', 'required', 'size'].forEach((p) => this.requestUpdate(p));
```
[the remark this answers is not in the document]
Comment (MarioCastigliano) [reply src/components/checkbox/checkbox/checkbox.ts:53]: Panel version has no size (for now), would it still make sense to move it in the mixin?

File: src/components/selection-expansion-panel/selection-expansion-panel.scss [hunk 24]
```diff
@@ -0,0 +1,201 @@
...
+  --sbb-selection-expansion-panel-border-radius: var(--sbb-border-radius-4x);
+
+  // As the selection panel has always a white/milk background, we have to fix the focus outline color
+  // to default color for cases where the selection panel is used in a negative context.
+  --sbb-focus-outline-color: var(--sbb-focus-outline-color-default);
+
+  display: contents;
+}
+
+:host([color='milk']) {
```
[the remark this answers is not in the document]
Comment (MarioCastigliano) [reply src/components/selection-expansion-panel/selection-expansion-panel.scss:35]: Should we ask @mcilurzo or just go with it?

File: src/components/checkbox/checkbox/checkbox.ts (hunk 23 above)
[the remark this answers is not in the document]
Comment (jeripeierSBB) [reply src/components/checkbox/checkbox/checkbox.ts:55]: I think the size question should be solved first and then I think again :)

File: src/components/checkbox/checkbox/checkbox.ts (hunk 22 above)
[the remark this answers is not in the document]
Comment (jeripeierSBB) [reply src/components/checkbox/checkbox/checkbox.ts:44]: so the SbbSlotStateController is only there because of the icon?

File: src/components/selection-expansion-panel/selection-expansion-panel.scss (hunk 24 above)
[the remark this answers is not in the document]
Comment (jeripeierSBB) [reply src/components/selection-expansion-panel/selection-expansion-panel.scss:35]: Please check with @mcilurzo

File: src/components/core/mixins/form-associated-checkbox-mixin.ts (hunk 20 above)
Comment (jeripeierSBB) [reply src/components/core/mixins/form-associated-checkbox-mixin.ts:83]: check "toggle-check", we solved it with willupdate() there. I think either you should remove it from toggle-check or, like in toggle-check, set data-checked in checkbox-panel. I'd prefer to have it in checkbox-panel.

```ts
  protected override async willUpdate(changedProperties: PropertyValues<this>): Promise<void> {
    super.willUpdate(changedProperties);

    if (changedProperties.has('checked')) {
      this.internals.ariaChecked = `${this.checked}`;
      this.toggleAttribute('data-checked', this.checked);
    }
  }
```

File: src/components/selection-expansion-panel/readme.md [hunk 25]
```diff
@@ -1,7 +1,7 @@
-The `sbb-selection-panel` component wraps either a [sbb-checkbox](/docs/components-sbb-checkbox-sbb-checkbox--docs)
+The `sbb-selection-expansion-panel` component wraps either a [sbb-checkbox](/docs/components-sbb-checkbox-sbb-checkbox--docs)
```
Comment (kyubisation) [inline src/components/selection-expansion-panel/readme.md:1]: Should reference the panel variants.

**Review by kyubisation — COMMENTED** [review]
Nice work! 😃

File: src/components/checkbox/checkbox-panel/checkbox-panel.e2e.ts [hunk 26]
```diff
@@ -0,0 +1,25 @@
+import { assert } from '@open-wc/testing';
+import { html } from 'lit/static-html.js';
+
+import { fixture } from '../../core/testing/private.js';
+
+import { SbbCheckboxPanelElement } from './checkbox-panel.js';
+
+describe(`sbb-checkbox-panel with ${fixture.name}`, () => {
```
Comment (jeripeierSBB) [inline src/components/checkbox/checkbox-panel/checkbox-panel.e2e.ts:8]: Maybe write a a comment about where the tests are

File: src/components/radio-button/radio-button-panel/radio-button-panel.ts [hunk 27]
```diff
@@ -0,0 +1,78 @@
...
+import { SbbRadioButtonCommonElementMixin, radioButtonCommonStyle } from '../common.js';
+
+import radioButtonPanelStyle from './radio-button-panel.scss?lit&inline';
+
+/**
+ /**
+ * It displays a radio button enhanced with the selection panel design.
+ *
+ * @slot - Use the unnamed slot to add content to the radio label.
+ * @slot subtext - Slot used to render a subtext under the label (only visible within a `sbb-selection-expansion-panel`).
```
Comment (jeripeierSBB) [inline src/components/radio-button/radio-button-panel/radio-button-panel.ts:16]:
```suggestion
 * @slot subtext - Slot used to render a subtext under the label (only visible within a `sbb-selection-expansion-panel`).
```
```suggestion
 * @slot subtext - Slot used to render a subtext under the label.
```

File: src/components/radio-button/radio-button-panel/radio-button-panel.ts [hunk 28]
```diff
@@ -0,0 +1,78 @@
...
+ * @slot suffix - Slot used to render additional content after the label (only visible within a `sbb-selection-expansion-panel`).
```
Comment (jeripeierSBB) [inline src/components/radio-button/radio-button-panel/radio-button-panel.ts:17]:
```suggestion
 * @slot suffix - Slot used to render additional content after the label.
```

File: src/components/radio-button/radio-button-panel/radio-button-panel.ts [hunk 29]
```diff
@@ -0,0 +1,78 @@
...
+import { SbbSlotStateController } from '../../core/controllers.js';
+import { EventEmitter } from '../../core/eventing.js';
+import { SbbPanelMixin, SbbUpdateSchedulerMixin } from '../../core/mixins.js';
+import { SbbRadioButtonCommonElementMixin, radioButtonCommonStyle } from '../common.js';
+
+import radioButtonPanelStyle from './radio-button-panel.scss?lit&inline';
+
+/**
+ /**
+ * It displays a radio button enhanced with the selection panel design.
```
Comment (jeripeierSBB) [inline src/components/radio-button/radio-button-panel/radio-button-panel.ts:13]:
```suggestion
 * It displays a radio button enhanced with the panel design.
```

File: src/components/radio-button/radio-button-panel/radio-button-panel.stories.ts [hunk 30]
```diff
@@ -0,0 +1,152 @@
...
+  borderless: false,
+  size: size.options![0],
+};
+
+const DefaultTemplate = ({ labelBoldClass, ...args }: Args): TemplateResult =>
+  html`<sbb-radio-button-panel ${sbbSpread(args)}
+    >${labelBoldClass ? html`<span class="sbb-text--bold">Label</span>` : 'Label'}
+    <span slot="subtext">Subtext</span>
+    <span slot="suffix" style="margin-inline-start: auto;">
+      <span style="display:flex;align-items:center;">
```
[the remark this answers is not in the document]
Comment (jeripeierSBB) [reply src/components/radio-button/radio-button-panel/radio-button-panel.stories.ts:88]: maybe add icon as well like in checkbox panel

File: src/components/radio-button/radio-button-panel/radio-button-panel.stories.ts [hunk 31]
```diff
@@ -0,0 +1,152 @@
...
+
+const meta: Meta = {
+  decorators: [
+    (story) => html` <div style="padding: 2rem;">${story()}</div> `,
+    withActions as Decorator,
+  ],
+  parameters: {
+    actions: {
+      handles: [
+        SbbRadioButtonPanelElement.events.radioButtonLoaded,
```
Comment (jeripeierSBB) [inline src/components/radio-button/radio-button-panel/radio-button-panel.stories.ts:138]: maybe remove it?

File: src/components/core/styles/mixins/selection-panel.scss [hunk 32]
```diff
@@ -0,0 +1,61 @@
+@use './a11y';
+
+@mixin selection-panel-variables {
+  --sbb-selection-expansion-panel-background: var(--sbb-color-white);
+  --sbb-selection-expansion-panel-border-color: var(--sbb-color-cloud);
+  --sbb-selection-expansion-panel-border-radius: var(--sbb-border-radius-4x);
+  --sbb-selection-expansion-panel-border-width: var(--sbb-border-width-1x);
+  --sbb-selection-expansion-panel-input-padding: var(--sbb-spacing-responsive-xs)
+    var(--sbb-spacing-responsive-xxs);
+  --sbb-selection-expansion-panel-animation-duration: var(--sbb-animation-duration-4x);
```
Comment (jeripeierSBB) [inline src/components/core/styles/mixins/selection-panel.scss:10]: Fallback to --sbb-disable-animation-zero-time needed: 
```suggestion
  --sbb-selection-expansion-panel-animation-duration: var(--sbb-disable-animation-zero-time, var(--sbb-animation-duration-4x));
```

File: src/components/selection-expansion-panel/selection-expansion-panel.scss [hunk 33]
```diff
@@ -0,0 +1,201 @@
...
+$open-anim-rows-to: 1fr;
+$open-anim-opacity-from: 0;
+$open-anim-opacity-to: 1;
+
+:host {
+  --sbb-selection-expansion-panel-cursor: pointer;
+  --sbb-selection-expansion-panel-background: var(--sbb-color-white);
+  --sbb-selection-expansion-panel-border-color: var(--sbb-color-cloud);
+  --sbb-selection-expansion-panel-animation-duration: var(
+    --sbb-disable-animation-time,
```
Comment (jeripeierSBB) [inline src/components/selection-expansion-panel/selection-expansion-panel.scss:17]:
```suggestion
    --sbb-disable-animation-zero-time,
```

**Review by jeripeierSBB — APPROVED** [review]
Technical approval. Great! :)

File: src/elements/radio-button/radio-button-group/radio-button-group.ts [hunk 34]
```diff
@@ -147,8 +147,10 @@ export class SbbRadioButtonGroupElement extends SbbDisabledMixin(LitElement) {
       },
     );
     this.addEventListener('keydown', (e) => this._handleKeyDown(e), { signal });
-    this._hasSelectionPanel = !!this.querySelector?.('sbb-selection-expansion-panel');
-    this.toggleAttribute('data-has-selection-expansion-panel', this._hasSelectionPanel);
+    this._hasPanelElement = !!this.querySelector?.(
+      'sbb-selection-expansion-panel, sbb-radio-button-panel',
+    );
+    this.toggleAttribute('data-has-selection-expansion-panel', this._hasPanelElement);
```
Comment (jeripeierSBB) [inline src/elements/radio-button/radio-button-group/radio-button-group.ts:153]: should the data attribute be renamed?

Comment (jeripeierSBB) [reply src/elements/radio-button/radio-button-group/radio-button-group.ts:153]: or should it check for checkbox panels too? or why not?

File: src/elements/checkbox/checkbox-panel/checkbox-panel.stories.ts [hunk 35]
```diff
@@ -0,0 +1,215 @@
...
+};
+
+export const withForm: StoryObj = {
+  render: TemplateWithForm,
+  argTypes: defaultArgTypes,
+  args: defaultArgs,
+};
+
+const meta: Meta = {
+  decorators: [withActions as Decorator],
```
Comment (jeripeierSBB) [inline src/elements/checkbox/checkbox-panel/checkbox-panel.stories.ts:206]: The decorator withActions is used but no events are listened

File: src/elements/checkbox/checkbox-panel/checkbox-panel.ts [hunk 36]
```diff
@@ -0,0 +1,133 @@
...
+      if (this.disabled !== changedProperties.get('disabled')!) {
+        this.stateChange.emit({ type: 'disabled', disabled: this.disabled });
+      }
+    }
+  }
+
+  protected override render(): TemplateResult {
+    return html`
+      <span class="sbb-selection-panel">
+        <div class="sbb-selection-expansion-panel__badge">
```
Comment (jeripeierSBB) [inline src/elements/checkbox/checkbox-panel/checkbox-panel.ts:102]: I'd vote for renaming to selection-panel (removing expansion) like in the root div
